frontend/components/videoPlayer.vue

```diff
@@ -0,0 +1,46 @@
+<script setup>
+    let sourceBuffer;
+    //this is temporary just to pass the check, We will want this to be set dynamically
+    let codec = 'video/mp4; codecs="avc1.4D401F"';
+ 
+	onMounted(()=>{
+        if(!MediaSource.isTypeSupported(codec)){
+            console.log(`codec: ${codec} is not supported`);
+            return;
+        }
+        sourceBuffer = new MediaSource();
+
+        sourceBuffer.addEventListener('sourceopen',(ev)=>{
```

Comment:
I don't really know what this part is. I mainly do not understand what ev or b is. But I am also super inexperienced. So it may be the fault of my own stupidity.

Reply:
If you dont understand something and need to ask for an explanation then you should so that you can understand and I can flush out the idea by explaining it to another person. So you should not feel bad about not knowing something. (Also there are so many weird edges of coding that you are going to find a lot of weird things that you wont understand)

Reply:
@Razzelz What is happening here is when the webpage is loaded I am creating a new media source that later on will be what holds the video data and the video player will pull all of the data out of the this buffer to play the video. And there are a couple of bugs in there that I did not remove before I pushed up